systemtest/src/main/java/io/strimzi/systemtest/performance/utils/UserOperatorPerformanceUtils.java

```diff
@@ -113,8 +124,171 @@ public static List<KafkaUser> getListOfKafkaUsers(final TestStorage testStorage,
+        synchronized (RESOURCE_CREATION_LOCK) {
+            KubeResourceManager.get().createResourceWithoutWait(listOfUsers.toArray(new KafkaUser[listOfUsers.size()]));
+        }
+
+        // Wait for each specific user in the list to be ready
+        listOfUsers.forEach(kafkaUser ->
+            KafkaUserUtils.waitForKafkaUserReady(testStorage.getNamespaceName(), kafkaUser.getMetadata().getName())
+        );
+    }
+
+    /**
+     * Manages the full lifecycle of Kafka users concurrently using a cached thread pool.
+     * This method processes creation, modification, and deletion for each user in separate threads,
+     * allowing unlimited concurrent operations which is optimal for I/O-bound Kubernetes API calls.
+     *
+     * @param testStorage           An instance of TestStorage containing configuration and state needed for user operations.
+     * @param numberOfUsers         The number of Kafka users to be processed.
+     * @param spareEvents           The number of spare events to be consumed during the process.
+     * @param warmUpTasksToProcess  The number of tasks to warm-up performance and optimize JIT. This number is used just for offsetting.
+     *
+     * @return                      The total time taken to complete all user lifecycles in milliseconds.
+     */
+    public static long processAllUsersConcurrently(TestStorage testStorage, int numberOfUsers, int spareEvents, int warmUpTasksToProcess) {
+        if (executorService.isShutdown() || executorService.isTerminated()) {
+            executorService = Executors.newCachedThreadPool();
+            LOGGER.info("Reinitialized ExecutorService for new test run.");
+        }
+
+        List<CompletableFuture<Void>> futures = new ArrayList<>();
+        ExtensionContext extensionContext = KubeResourceManager.get().getTestContext();
+
+        long startTime = System.nanoTime();
+
+        for (int userIndex = warmUpTasksToProcess; userIndex < numberOfUsers + warmUpTasksToProcess; userIndex++) {
+            final int startIndex = userIndex;
+            CompletableFuture<Void> future = CompletableFuture.runAsync(() -> performFullLifecycle(startIndex,  testStorage, extensionContext), executorService);
+            futures.add(future);
+        }
+
+        // consume spare events
+        for (int j = 0; j < spareEvents; j++) {
+            futures.add(j, CompletableFuture.completedFuture(null));
+        }
+
+        // Wait for all users to complete their lifecycle
+        CompletableFuture.allOf(futures.toArray(new CompletableFuture[0])).join();
+        LOGGER.info("All user lifecycles completed.");
+
+        long allTasksTimeMs = Duration.ofNanos(System.nanoTime() - startTime).toMillis();
+
+        if (warmUpTasksToProcess != 0) {
+            // boundary between tests => less likelihood that tests would influence each other
+            LOGGER.info("Cooling down");
+            try {
+                Thread.sleep(5_000);
```

Comment:
Could you use `LockSupport.parkNanos()` that should be thread-safe?

Reply:
That piece of code is sequential; the whole async stuff effectively ends at [1], so I don't see a clear benefit in adding it. 

[1] - https://github.com/strimzi/strimzi-kafka-operator/pull/12091/files#diff-039111f3d2145c68a8942d19c264dfa607f3c2a6bab5ce712c507f89d11780a7R168